lib/src/visitor/any_selector.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:meta/meta.dart';
+
+import '../ast/selector.dart';
+import 'interface/selector.dart';
+
+/// A visitor that visits each selector in a Sass selector AST and returns
+/// `true` if any of the individual methods return `true`.
+///
+/// Each method returns `false` by default.
+@internal
+abstract class AnySelectorVisitor implements SelectorVisitor<bool> {
+  const AnySelectorVisitor();
+
+  bool visitComplexSelector(ComplexSelector complex) => complex.components
+      .any((component) => visitCompoundSelector(component.selector));
```

Comment:
Is there a reason why this calls `visitCompoundSelector` directly instead of `component.selector.accept(this)`?